### ISSUE TITLE
Keyboard accessibility improvement on `TileSelect` 

### ISSUE THREAD

**Issue by JinCoreana** [Sage/carbon#5932]
### Desired behaviour

When `customActionButton` is specified with an icon, remove focus from `div` to avoid two focuses in the same tile when navigating by Tab keyboard.

### Current behaviour

 When navigating by Tab key, I see 2 focuses within the same item and this can not be resolved by adding tabIndex as it does not exist in the type definition.

### Suggested Solution

https://codesandbox.io/s/practical-feather-97xfts?file=/src/App.js 

### CodeSandbox or Storybook URL

https://t8stde.csb.app

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by DipperTheDan** [comment]
@tempertemper - What is you advice on this issue regarding keyboard navigation in Tile Select?

**Comment by tempertemper** [comment]
If the whole tile is focusable, all of the information in the tile should be read out to a screen reader user, and as labels need to indicate *what the button will do* when pressed this would fail [2.4.6 Headings and Labels](https://www.w3.org/TR/WCAG21/#headings-and-labels) since the label gives information about that the thing *is*.

(At least, that was my expectation before firing up my screen reader – I actually get told "1, unticked tickbox" instead of the info in the box, which is odd…)

There's also a danger of mis-clicks by users who don't realise the whole tile is clickable (since there is no visual affordance to suggest this).

So focusing only on the button that will carry out the action is the right thing to do 👍 

There's a bit more to this, which may require another ticket – this is not a radio group, but it is being identified as a radio group programmatically. This is essentially a bunch of tiles that are selected by pressing a button, and the `aria-pressed` attribute alongside clear labelling that a tile has been selected or not is required. Definitely one to collaborate with a designer on, and I'm happy to provide guidance.

**Comment by ljemmo** [comment]
we're actually looking to deprecate this component in the DS as it was originally created for in-product subscription selections. The new subscription tile component supersedes this. @JinCoreana - where were you hoping to use this component are you able to send me a design spec?

**Comment by JinCoreana** [comment]
@ljemmo Sure, we use this component to display sidebar options, such as the attached video. 
It works perfectly when users click on it with a mouse, but when users navigate using the tab key, the focus is duplicated on the same element and also the focus on the whole tile does not trigger any action, causing confusion for users. So I wanted to resolve this issue to improve accessibility. Any suggestions would be much appreciated.

https://user-images.githubusercontent.com/90848080/231684475-639bbf94-a57f-46c7-a505-44a29f43c46f.mov

**Comment by tempertemper** [comment]
@ljemmo Ah it does feel like a different use case! These are just plain old buttons that are styled in a certain way, including a decorative right-pointing chevron. Nothing form-like about them, which is kind of where the tile select is usually used: press a button, move across / drill down to the previously hidden panel on its right.

**Comment by Parsium** [comment]
@ljemmo Ah that makes sense 👍🏼 

Unfortunately we can't disable the box shadowing on `Card` currently. It should be [possible to use `Tile` as shown here](https://codesandbox.io/s/condescending-field-95gu6o?file=/src/App.tsx) to create a similar(ish) pattern, but I appreciate that component is designed for containing static content only, so it may be better that a new component is introduced for this particular use case.

**Comment by ljemmo** [comment]
@Parsium what if we provide a token for 0 box shadow values? This way we can pass this into the card comp. I agree that this shouldn't be built in tile as tiles are static.

**Comment by JinCoreana** [comment]
@Parsium Sure, I believe we could find a solution by using `Box`. However this was just me looking into improving the accessibility and it has not ben reported as an issue yet. So, we could leave it and discuss adjusting the design if it becomes an issue in the future. 

 However, I wonder if it would be a big deal to add tabIndex attribute in `TileSelect` component? I could workaround it as [attached link]( https://t8stde.csb.app/). However, I am currently unable to adopt this solution as it throws a type error. 

 Additionally, the example you provided with the `Tile` component example could be a viable alternative solution without revising the current design, provided that we find a way to remove the vertical border between items.

**Comment by Parsium** [comment]
Thanks for this @JinCoreana, unfortuntately introducing `tabIndex` as a prop would introduce further problems. Since `TileSelect` was designed to act like a group of radio buttons, where one of the options can be selected, removing the ability to focus the tile would mean [patterns like this would break for keyboard users](https://codesandbox.io/s/elated-sutherland-evuqje?file=/src/App.tsx).

I agree with @ljemmo that ultimately a new component would be better suited for this pattern you are working with, but to temporarily address the accessibility issue you are working on, I would suggest using `Tile` in the mean time. While this isn't what `Tile` is designed for as @ljemmo mentioned, we have had examples of interactive controls in `Tile` in our Carbon docs for some time now (which we ultimately intend to address to better align with Design System).

However if this is strongly discouraged from a design point of view, I think raising a separate ticket to amend the design of the option list would need raising anyway.

Hope this helps?

**Comment by JinCoreana** [comment]
Thanks so much @Parsium for your advice. I understood. Don't worry, I will close this issue for now. And in any case, I will workaround it using `Tile` as you suggested. Thank you! :)
